app/models/conversation.rb

```diff
@@ -61,6 +61,7 @@ class Conversation < ApplicationRecord
 
   has_many :messages, dependent: :destroy, autosave: true
   has_one :csat_survey_response, dependent: :destroy
+  has_many :notifications, as: :notification_primary_actor, dependent: :destroy
```

Comment:
```suggestion
  has_many :notifications, as: :primary_actor, dependent: :destroy
```

should be suffice. Please test before applying the change.